cmds/watch.c

```diff
@@ -97,6 +97,37 @@ static void cmd_watch_list(struct watchman_client *client, json_t *args)
 }
 W_CMD_REG("watch-list", cmd_watch_list, CMD_DAEMON, NULL)
 
+/* watch-purge
+ * Stops watching all roots */
+static void cmd_watch_purge(struct watchman_client *client, json_t *args)
+{
+  json_t *resp;
+  json_t *purged;
+  json_t *root_paths;
+  size_t  root_paths_size, i;
+  unused_parameter(args);
+
+  root_paths = w_root_watch_list_to_json();
```

Comment:
Could you add a `w_root_stop_all()` function to root.c that combines the iteration and the stop requests under the root_lock mutex?  That way this command becomes more atomic.  The `w_root_stop_all` function could return a json_t array listing the names that it stopped